cdap-ui/app/hydrator/templates/partial/node-config-modal/node-config/plugin-edit-properties-template.html

```diff
@@ -41,107 +41,29 @@ <h4>{{::group.display}}</h4>
-              -->
-              <div ng-if="(['javascript-editor', 'python-editor', 'scala-editor'].indexOf(field['widget-type']) !== -1)">
-                <div data-name="field"
-                     class="my-widget-container"
-                     ng-class="{'select-wrapper': field.widget === 'select'}"
-                     data-model="HydratorPlusPlusNodeConfigCtrl.state.node.plugin.properties[field.name]"
-                     data-myconfig="field"
-                     disabled="isDisabled"
-                     data-properties="HydratorPlusPlusNodeConfigCtrl.state.node.plugin.properties"
-                     widget-disabled="HydratorPlusPlusNodeConfigCtrl.state.node.pluginTemplate && HydratorPlusPlusNodeConfigCtrl.state.node.lock[field.name]"
-                     widget-container>
-                </div>
+          <div class="form-group">
+            <div class="col-xs-offset-8 col-xs-4">
+                <span ng-if="field['plugin-function'] && (!field['plugin-function']['position'] || field['plugin-function']['position'] === 'top') && !isDisabled">
+                  <plugin-functions fn-config="field['plugin-function']" node="HydratorPlusPlusNodeConfigCtrl.state.node">
+                  </plugin-functions>
+                </span>
               </div>
-
-              <span class="badge macro-indicator pull-right"
-                    ng-if="HydratorPlusPlusNodeConfigCtrl.state.node._backendProperties[field.name].macroSupported"
-                    uib-tooltip="A placeholder that will be substituted at runtime. eg. ${placeholder} or ${secure(key)}"
-                    tooltip-append-to-body="true"
-                    tooltip-popup-delay="300"
-                    tooltip-placement="left">
-                Macro
-              </span>
-            </div>
-            <div class="col-xs-1 info-attr">
-              <span class="fa fa-info-circle" uib-tooltip="{{::field.description}}" tooltip-placement="right" tooltip-append-to-body="true"></span>
-              <span
-                class="text-danger fa fa-exclamation-triangle"
-                uib-tooltip="{{field.errorTooltip}}"
-                tooltip-append-to-body="true"
-                tooltip-placement="right"
-                tooltip-class="error-tooltip error-node-config"
-                ng-if="!HydratorPlusPlusNodeConfigCtrl.state.node.warning && HydratorPlusPlusNodeConfigCtrl.state.node._backendProperties[field.name].required && !HydratorPlusPlusNodeConfigCtrl.state.node.plugin.properties[field.name]"
-              >
-              </span>
-            </div>
           </div>
-        <div class="form-group">
-          <div class="col-xs-offset-8 col-xs-4">
+
+          <macro-widget-toggle
+            node="HydratorPlusPlusNodeConfigCtrl.state.node"
+            field="field"
+            disabled="isDisabled"></macro-widget-toggle>
```

Comment:
here too

Reply:
this is not react, cannot use self closing tag

Reply:
oh forgot lol :joy: